src/entity/user/UserSkill.ts

```diff
@@ -0,0 +1,28 @@
+import { Column, Entity, ManyToMany, PrimaryColumn } from 'typeorm';
+import type { UserExperience } from './experiences/UserExperience';
+
+@Entity()
+export class UserSkill {
+  @PrimaryColumn({
+    type: 'text',
+    update: false,
+    insert: false,
+    nullable: false,
+    unique: true,
+    generatedType: 'STORED',
+    asExpression: `trim(BOTH '-' FROM regexp_replace(lower(trim(COALESCE(LEFT(name,100),''))), '[^a-z0-9-]+', '-', 'gi'))`,
+  })
+  slug: string;
+
+  @Column({ type: 'text', unique: true })
+  name: string;
+
+  @Column({ type: 'text', nullable: true })
+  description: string;
+
+  @ManyToMany(
+    'UserExperience',
+    (experience: UserExperience) => experience.skills,
+  )
+  experiences: Promise<UserExperience[]>;
+}
```

Comment:
Entities look good, please just verify all have cascades so that everything is cleaned up when user, skill and so on is deleted